src/components/jar_details/UtxoList.tsx

```diff
@@ -0,0 +1,270 @@
+import React, { useMemo } from 'react'
+import * as rb from 'react-bootstrap'
+import classnames from 'classnames'
+import { useTranslation } from 'react-i18next'
+import { Table, Header, HeaderRow, HeaderCell, Body, Row, Cell } from '@table-library/react-table-library/table'
+import { useRowSelect, HeaderCellSelect, CellSelect, SelectTypes } from '@table-library/react-table-library/select'
+import { useSort, HeaderCellSort } from '@table-library/react-table-library/sort'
+import * as TableTypes from '@table-library/react-table-library/types/table'
+import { useTheme } from '@table-library/react-table-library/theme'
+import { useSettings } from '../../context/SettingsContext'
+import { Utxo, WalletInfo } from '../../context/WalletContext'
+import * as fb from '../fb/utils'
+import Sprite from '../Sprite'
+import Balance from '../Balance'
+import styles from './UtxoList.module.css'
+
+const ADDRESS_STATUS_COLORS: { [key: string]: string } = {
+  new: 'normal',
+  used: 'normal',
+  reused: 'danger',
+  'cj-out': 'success',
+  'change-out': 'warning',
+  'non-cj-change': 'normal',
+  deposit: 'normal',
+}
+
+const SORT_KEYS = {
+  frozenOrLocked: 'FROZEN_OR_LOCKED',
+  value: 'VALUE',
+}
+
+const TABLE_THEME = {
+  Table: `
+    --data-table-library_grid-template-columns: 2rem 3.5rem 2fr 4fr 4rem 3fr 1fr;
+    font-size: 0.9rem;
+  `,
+  BaseCell: `
+    &:nth-of-type(1) {
+      text-align: center;
+    }
+  `,
+  Cell: `
+    &:nth-of-type(2) {
+      text-align: center;
+    }
+    &:nth-of-type(5) > div {
+      display: flex;
+      justify-content: center;
+    }
+  `,
+  Row: `
+    &.row-select-selected {
+      color: inherit;
+      font-weight: inherit;
+    }
+  `,
+}
+
+interface UtxoListProps {
+  utxos: Array<Utxo>
+  walletInfo: WalletInfo
+  setSelectedUtxoIds: (selectedUtxoIds: Array<string>) => void
+  setDetailUtxo: (utxo: Utxo) => void
+}
+
+const UtxoList = ({ utxos, walletInfo, setSelectedUtxoIds, setDetailUtxo }: UtxoListProps) => {
+  const { t } = useTranslation()
+  const settings = useSettings()
+
+  const toUtxo = (tableNode: TableTypes.TableNode): Utxo => {
+    const { id, ...utxo } = tableNode
+
+    return utxo as Utxo
+  }
+
+  const utxoTags = (utxo: Utxo, walletInfo: WalletInfo) => {
+    var rawStatus = walletInfo.addressSummary[utxo.address]?.status
+
+    var status: string | null = null
+    var locktime: string | null = null
+
+    if (rawStatus) {
+      const indexOfLockedTag = rawStatus.indexOf('[LOCKED]')
+
+      if (indexOfLockedTag !== -1) {
+        locktime = rawStatus.substring(0, indexOfLockedTag).trim()
```

Comment:
`utxo.locktime` is also populated for time-locked utxos. Is there a specific reason why locktime is parsed from the `status` value? Just asking, because It has not been _that_ reliable in the past.

Reply:
That is a good point actually. Let me change that. 👍